### PR TITLE
Patrol point render

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #2779

# Changes proposed in this pull request:
- Render patrol points
- Fix citizen crash
- enable banner rendering

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/structures/client/TemplateRenderHandler.java [hunk 1]
```diff
@@ -57,10 +125,78 @@ public void draw(final Template template, final Rotation rotation, final Mirror
...
+            return;
+        }
+        final EntityPlayer perspectiveEntity = Minecraft.getMinecraft().player;
+        final double interpolatedEntityPosX = perspectiveEntity.lastTickPosX + (perspectiveEntity.posX - perspectiveEntity.lastTickPosX) * partialTicks;
+        final double interpolatedEntityPosY = perspectiveEntity.lastTickPosY + (perspectiveEntity.posY - perspectiveEntity.lastTickPosY) * partialTicks;
+        final double interpolatedEntityPosZ = perspectiveEntity.lastTickPosZ + (perspectiveEntity.posZ - perspectiveEntity.lastTickPosZ) * partialTicks;
+
+        if (template != this.template)
+        {
+            this.template = template;
```
Comment (marchermans) [inline src/main/java/com/minecolonies/structures/client/TemplateRenderHandler.java]: What is the reason for caching this? Isn't this method called every tick after draw anyway?

**Comment by Raycoms** [comment]
I was caching this for the general schematics, but I wanted to make it work
for all tileEntities but it didn't work for beds, so I can remove the
caching as well. There are not enough banners in schematics in general to
cache this.

2018-08-13 11:43 GMT-03:00 Marc Hermans <notifications@github.com>:

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count |  New  | Old  | Diff  |
| ------ |:-----:|:-----:|:----:|:-----:|
| Total  | 39999 | 15522 | 8607 | +6915 |
| Errors |  597  |  32   |  19  |  +13  |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=15854&tab=Inspection&buildTypeId=bt88)

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 39993 |  2  |  8  |  -6  |
| Errors |  597  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=16065&tab=Inspection&buildTypeId=bt88)

**Review by marchermans — APPROVED** [review]

**Review by LDTTeam-Buildsystem — APPROVED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 39995 |  7  |  5  |  +2  |
| Errors |  597  |  0  |  0  |  +0  |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=16129&tab=Inspection&buildTypeId=bt88)

**Pull request merged**